### PR TITLE
applications: asset_tracker_v2: Handle lwm2m FOTA state changes

### PR DESCRIPTION
**Pull request by simensrostad**
Handle lwm2m FOTA state changes and propagate the correct FOTA event to
the rest of the application. This allows the application to display
the correct FOTA LED patterns, and to detect when the FOTA has been
performed and a reboot is required.

Add Kconfig option `CONFIG_LWM2M_CLIENT_UTILS_FIRMWARE_UPDATE_REBOOT`
that gives the application the opportunity to disable
rebooting from the firmare object when it wants to handle
the reboot it self.

Update LWM2M integration test.

Fixes CIA-644

### CODE REVIEW

**Comment by simensrostad** [comment]
PR is ready for review now.

**Comment by simensrostad** [comment]
@balaji-nordic Updated the test for the lwm2m integration layer. Nice if you could have a look!

File: applications/asset_tracker_v2/src/cloud/lwm2m_integration/lwm2m_integration.c [hunk 1]
```diff
@@ -251,6 +251,70 @@ static int modem_mode_request_cb(enum lte_lc_func_mode new_mode, void *user_data
...
+	ARG_UNUSED(obj_inst_id);
+	ARG_UNUSED(args);
+	ARG_UNUSED(args_len);
+
+	int err;
+	struct cloud_wrap_event cloud_wrap_evt = {
+		.type = CLOUD_WRAP_EVT_FOTA_DONE
+	};
+
+	err = lwm2m_firmware_apply_update(0);
```
Comment (jarlamsa) [inline applications/asset_tracker_v2/src/cloud/lwm2m_integration/lwm2m_integration.c:308]: You could also pass the `obj_inst_id` here, it doesn't really matter currently as the multi-instanced FOTA isn't used, but might save some debugging time in future when the multi-instanced FOTA is defined.

**Review by jarlamsa — APPROVED** [review]
LGTM, one comment about object instances.

File: applications/asset_tracker_v2/src/cloud/lwm2m_integration/lwm2m_integration.c (hunk 1 above)
Comment (simensrostad) [reply applications/asset_tracker_v2/src/cloud/lwm2m_integration/lwm2m_integration.c:308]: Sounds good, ill make sure to update the input arg.

File: subsys/net/lib/lwm2m_client_utils/fota/settings.c [hunk 2]
```diff
@@ -66,6 +66,11 @@ static struct settings_handler fota_settings = {
 int fota_settings_init(void)
 {
 	int err;
+	static bool init;
```
Comment (maxd-nordic) [inline subsys/net/lib/lwm2m_client_utils/fota/settings.c:69]: This is just to prevent calling `settings_subsys_init` multiple times, right?

**Review by maxd-nordic — APPROVED** [review]

File: subsys/net/lib/lwm2m_client_utils/fota/settings.c (hunk 2 above)
Comment (simensrostad) [reply subsys/net/lib/lwm2m_client_utils/fota/settings.c:69]: Yes.

**Review by VeijoPesonen — APPROVED** [review]

**Review by SeppoTakalo — APPROVED** [review]

File: applications/asset_tracker_v2/src/cloud/lwm2m_integration/lwm2m_integration.c [hunk 3]
```diff
@@ -251,6 +251,69 @@ static int modem_mode_request_cb(enum lte_lc_func_mode new_mode, void *user_data
...
 }
 
+static int firmware_update_state_cb(uint8_t update_state)
+{
+	int err;
+	uint8_t update_result;
+	struct cloud_wrap_event cloud_wrap_evt = { 0 };
+
+	/* Get the firmware object update result code */
+	err = lwm2m_engine_get_u8("5/0/5", &update_result);
```
Comment (balaji-nordic) [inline applications/asset_tracker_v2/src/cloud/lwm2m_integration/lwm2m_integration.c:261]: Use macro for 5/0/5 ? like you have done in the unit test?

File: applications/asset_tracker_v2/tests/lwm2m_integration/src/lwm2m_integration_test.c [hunk 4]
```diff
@@ -21,18 +19,27 @@
 #define LIFETIME_PATH			"1/0/1"
 #define REBOOT_PATH			"3/0/4"
 #define ENDPOINT_NAME_EXPECTED		":urn:id:test"
+#define FIRMWARE_UPDATE_RESULT_PATH	"5/0/5"
+#define OBJECT_INSTANCE_ZERO		0
```
Comment (balaji-nordic) [inline applications/asset_tracker_v2/tests/lwm2m_integration/src/lwm2m_integration_test.c:23]: We usually dont have macros for '0's. But perhaps this makes sense here?

**Review by balaji-nordic — APPROVED** [review]
Just a couple of minor comments which you may or may not address

**Pull request merged**